lambda/functions/schedule/check-black-list/handler.ts

```diff
@@ -0,0 +1,65 @@
+import { findAllBlackListItems, deleteEmailsFromBlacklist } from '../../../shared/services/dynamo.service';
+import {sendPayloadToSqs} from "../../../shared/services/sqs.service";
+import {BlackWhiteListEntity} from "../../../shared/interfaces";
+import {checkEmailDomain} from "../../../shared/utils";
+import {saveTSMessage} from "../../../shared/services/timestream.service";
+import {ValidationLogNames} from "../../../shared/enums";
+
+const oneWeekInMilliseconds = 7 * 24 * 60 * 60 * 1000;
+
+export const handler = async (): Promise<void> => {
+  try {
+    const oneWeekAgo = new Date(+(new Date()) - oneWeekInMilliseconds).toISOString();
+    const blackListItems = await findAllBlackListItems({
+      FilterExpression: 'createdAt < :oneWeekAgo',
+      ExpressionAttributeValues: {
+        ':oneWeekAgo': oneWeekAgo,
+      },
+    });
+
+    // dividing emails in the rehabilitate and check groups
+    const [itemsNeedToRehabilitate, itemsNeedToCheck] = blackListItems.reduce(([a, b], item) => {
+      const rehabilitatedEmailDomains = (process.env.REHABILITATED_EMAIL_DOMAINS!).split(',');
```

Comment:
The code block from 22 to 26 is too redundant. 
The validation logic should be:
Filter black list of email by DOMAIN_BLACK_LIST and if domain of email not in this list go to review

///SELECT * FROM email_balck_list WHERE DOMAIN NOT IN (DOMAIN_BLACK_LIST)